File: src/main/java/org/cubectl/identity/role/PermissionRepository.java
package org.cubectl.identity.role;

import java.util.Collection;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;

public interface PermissionRepository extends JpaRepository<PermissionEntity, UUID> {

    Optional<PermissionEntity> findByCode(String code);

    List<PermissionEntity> findByCodeIn(Collection<String> codes);
}

File: src/main/kotlin/org/cubectl/identity/role/PermissionRepository.kt
package org.cubectl.identity.role

import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface PermissionRepository : JpaRepository<PermissionEntity, UUID> {

    fun findByCode(code: String): Optional<PermissionEntity>

    fun findByCodeIn(codes: Set<String>): List<PermissionEntity>
}
